core/model/configuration.py

```diff
@@ -1140,6 +1140,11 @@ def __get__(
                 logging.getLogger(self.__class__.__name__).error(
                     f"Could not typecast {self.label} value '{setting_value}'. Falling back to default"
                 )
+                if setting_value != "":
+                    # A non-empty value is a "bad" value, and should raise an exception
```

Comment:
I chose to fallback to default on empty because that's what the UI shows when if a value is deleted and saved. Eg: delete the max retries value in an overdrive collection and save. It will show up as `3` which is the default, whereas the value is empty.

Reply:
That makes sense to me. I still think we could move the check for an empty number value from the `except` block to just after the `if` block at ll. 1133-1134:
```
    if setting_value is None:
        setting_value = self.default
    elif self.type == ConfigurationAttributeType.NUMBER and setting_value == “”:
        setting_value = self.default
    …
```
(or the above could be combined into a single `if statement`)

But, if not, what's there should work fine.

Reply:
I was going for the "Ask for forgiveness" approach as opposed to the "Look before you leap" approach. Python advocates this approach if the number of times an exception is encountered is far lesser than the happy path.
Somebody did a [non-scientific benchmark](https://www.geeksforgeeks.org/try-except-vs-if-in-python/)